target/consensource-api
# Add JWT auth failure route

Signed-off-by: Patrick-Erichsen <Patrick.Erichsen@target.com>

## Proposed change/fix

This PR adds an additional endpoint to handle JWT auth failures. Without this endpoint, when JWT auth fails, there is nowhere to forward the request to so the client receives a generic 404 error.

Currently, the endpoint just returns a more descriptive 401 error with an explanation of the failure.

Also includes a few misc. logging updates.

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_


## How to run/test

- Open up `docker-compose.yaml` and make sure the API has an environment config with `OAUTH_VALIDATION_URL: "https://oauth.iam.perf.target.com"` (this is required to make the API attempt to perform JWT auth)
- `./docker-helper --build api`
- `./docker-helper --run api`
- Visit http://localhost:8080/index_standards_body.html#!/signUp and create an account
- Check the web console for an error that reads `"Bad request: JWT authentication failed"`. The docker-compose logs should also have the following:

```
INFO  | rocket::rocket:205   | POST /api/users application/json; charset=UTF-8:
INFO  | rocket::rocket:289   | Matched: POST /api/users application/json (create_user)
INFO  | rocket::rocket:297   | Outcome: Forward
INFO  | rocket::rocket:289   | Matched: POST /api/users [2] application/json (create_user_jwt_failure)
INFO  | rocket::rocket:297   | Outcome: Success
INFO  | rocket::rocket:120   | Response succeeded.
```
